src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSubject.cs

```diff
@@ -223,4 +225,81 @@ void CheckObjectValidity()
         //    return this.subjectKeyClaimset;
         //}
     }
+
+    internal class SamlSubjectEqualityComparer : IEqualityComparer<SamlSubject>
+    {
+        public bool Equals(SamlSubject subject1, SamlSubject subject2)
+        {
+            if (subject1 == null && subject2 == null)
+                return true;
+
+            if (subject1 == null | subject2 == null)
```

Comment:
Both of them work here. | still can check if one of them null but the other one is not.